app/public/cantusdata/management/commands/import_data.py

```diff
@@ -79,30 +80,41 @@ def handle(self, *args, **options):
     @transaction.atomic
     def import_manuscript_data(self, **options):
         self.stdout.write("Starting manuscript import process.")
+        cdb_base_url = "https://cantus.uwaterloo.ca/"
+        source_importer = SourceImporter(cdb_base_url)
+        source_ids = source_importer.request_source_ids()
         i = 0
-        for source, name in sources.items():
-            self.stdout.write(source + " " + name)
-            # Getting the fields from the scraper
-            metadata = parse_manuscript(source)
-            name = metadata.get("Title", "")
-            cantus_url = metadata.get("CantusURL", "")
-            csv_export_url = metadata.get("CSVExport", "")
-            siglum = metadata.get("Siglum", "")
-            date = metadata.get("Date", "")
-            provenance = metadata.get("Provenance", "")
-            description = metadata.get("Summary", "")
-            # Populating the Manuscript model
-            manuscript = Manuscript()
-            manuscript.name = name
-            manuscript.cantus_url = cantus_url
-            manuscript.csv_export_url = csv_export_url
-            manuscript.siglum = siglum
-            manuscript.date = date
-            manuscript.provenance = provenance
-            manuscript.description = description
-            manuscript.is_mapped = "UNMAPPED"
-            manuscript.save()
-            i += 1
+        for source_id in source_ids:
+            try:
+                source = source_importer.get_source_data(source_id)
+                # Getting the fields from the scraper
+                id = source.get("id")
+                name = source.get("name")
+                cantus_url = cdb_base_url + "source/" + id
+                csv_export_url = cdb_base_url + "sites/default/files/csv/" + id + ".csv"
+                siglum = source.get("siglum")
+                date = source.get("date")
+                provenance = source.get("provenance")
+                description = source.get("description")
+                # Populating the Manuscript model
+                manuscript = Manuscript()
+                manuscript.id = int(id)
+                manuscript.name = name
+                manuscript.cantus_url = cantus_url
+                manuscript.csv_export_url = csv_export_url
+                manuscript.siglum = siglum
+                manuscript.date = date
+                manuscript.provenance = provenance
+                manuscript.description = description
+                manuscript.is_mapped = "UNMAPPED"
```

Comment:
ah. i see line 9 in `/models/manuscript.py`. Maybe a more descriptive name for this field would be `manuscript.mapping_status`.

Reply:
Yes, this has to do with the history of this field: it used to be binary, but then we needed to expand it, but didn't change the name.

You are right that another name would be better, but I don't want to do that with this PR because it would involve changing a bunch of unrelated views and templates.